src/contract.js

```diff
@@ -88,49 +80,75 @@ class Contract {
       throw new Error('methodObj should not be undefined.');
     }
 
-    let dataHex = '';
-    dataHex = dataHex.concat(Encoder.objToHash(methodObj, true));
+    // function hash
+    const funcHash = Encoder.objToHash(methodObj, true);
 
-    let hex;
-    _.each(methodObj.inputs, (item, index) => {
+    const numOfParams = methodObj.inputs.length;
+
+    // create an array of data hex strings which will be combined at the end
+    const dataHexArr = _.times(numOfParams, _.constant(null));
+
+    // calculate start byte for dynamic data
+    let dataLoc = 0;
+    _.each(methodObj.inputs, (item) => {
       const type = item.type;
+      if (type.match(Constants.REGEX_STATIC_ARRAY)) {
+        // treat each static array as an individual slot for dynamic data location purposes
+        const arrCap = _.toNumber(type.match(Constants.REGEX_NUMBER)[1]);
+        dataLoc += arrCap;
+      } else {
+        dataLoc += 1;
+      }
+    });
 
-      if (type === 'address') {
-        hex = Encoder.addressToHex(args[index]);
-        dataHex = dataHex.concat(hex);
-      } else if (type === 'bool') {
-        hex = Encoder.boolToHex(args[index]);
-        dataHex = dataHex.concat(hex);
-      } else if (type.match(REGEX_UINT)) {
-        hex = Encoder.uintToHex(args[index]);
-        dataHex = dataHex.concat(hex);
-      } else if (type.match(REGEX_INT)) {
-        hex = Encoder.intToHex(args[index]);
-        dataHex = dataHex.concat(hex);
-      } else if (type.match(REGEX_BYTES_ARRAY)) { // fixed bytes array, ie. bytes32[10]
-        const arrCapacity = _.toNumber(type.match(REGEX_NUMBER)[1]);
-        if (args[index] instanceof Array) {
-          hex = Encoder.stringArrayToHex(args[index], arrCapacity);
-          dataHex = dataHex.concat(hex);
-        } else {
-          hex = Encoder.stringToHex(args[index], MAX_BYTES_PER_ARRAY_SLOT * arrCapacity);
-          dataHex = dataHex.concat(hex);
+    _.each(methodObj.inputs, (item, index) => {
+      const type = item.type;
+      let hex;
+
+      if (type === Constants.BYTES
+        || type === Constants.STRING
+        || type.match(Constants.REGEX_DYNAMIC_ARRAY)) { // dynamic types
+        let data = '';
+        if (type === Constants.BYTES) {
+          console.error('dynamics bytes conversion not implemented.');
+        } else if (type === Constants.STRING) {
```

Comment:
why not throw error directly using one if

Reply:
changed